# How to Select Columns using Python Pandas

In this notebook, we will look at different ways we can select columns from a master dataframe. This can be useful when creating a more relevant dataframe to undertake your analysis on. 

By the end of this notebook you will be armed with multiple ways to select columns for your dataframe. These include:
- Select columns by name
- Select columns by datatype
- Select columns by index
- Select columns by condition statements

## Getting ready
Let's start by importing pandas into our notebook

In [1]:
import pandas as pd

Next we will create our master dataframe by importing our dataset from a csv file.

In [2]:
df = pd.read_csv('data/players_22.csv', delimiter = ',')

C:\Users\pleas\AppData\Local\Temp\ipykernel_24464\110054706.py:1: DtypeWarning: Columns (25,108) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('data/players_22.csv', delimiter = ',')


Let's have a sneak peak at our dataframe

In [3]:
df.head()

sofifa_id                                         player_url  \
0     158023  https://sofifa.com/player/158023/lionel-messi/...   
1     188545  https://sofifa.com/player/188545/robert-lewand...   
2      20801  https://sofifa.com/player/20801/c-ronaldo-dos-...   
3     190871  https://sofifa.com/player/190871/neymar-da-sil...   
4     192985  https://sofifa.com/player/192985/kevin-de-bruy...   

          short_name                            long_name player_positions  \
0           L. Messi       Lionel Andrés Messi Cuccittini       RW, ST, CF   
1     R. Lewandowski                   Robert Lewandowski               ST   
2  Cristiano Ronaldo  Cristiano Ronaldo dos Santos Aveiro           ST, LW   
3          Neymar Jr        Neymar da Silva Santos Júnior          LW, CAM   
4       K. De Bruyne                      Kevin De Bruyne          CM, CAM   

   overall  potential    value_eur  wage_eur  age  ...   lcb    cb   rcb  \
0       93         93   78000000.0  320000.0   34  ...  50+3  50+3  50+3   
1       92         92  119500000.0  270000.0   32  ...  60+3  60+3  60+3   
2       91         91   45000000.0  270000.0   36  ...  53+3  53+3  53+3   
3       91         91  129000000.0  270000.0   29  ...  50+3  50+3  50+3   
4       91         91  125500000.0  350000.0   30  ...  69+3  69+3  69+3   

     rb    gk                                    player_face_url  \
0  61+3  19+3  https://cdn.sofifa.net/players/158/023/22_120.png   
1  61+3  19+3  https://cdn.sofifa.net/players/188/545/22_120.png   
2  60+3  20+3  https://cdn.sofifa.net/players/020/801/22_120.png   
3  62+3  20+3  https://cdn.sofifa.net/players/190/871/22_120.png   
4  75+3  21+3  https://cdn.sofifa.net/players/192/985/22_120.png   

                            club_logo_url  \
0  https://cdn.sofifa.net/teams/73/60.png   
1  https://cdn.sofifa.net/teams/21/60.png   
2  https://cdn.sofifa.net/teams/11/60.png   
3  https://cdn.sofifa.net/teams/73/60.png   
4  https://cdn.sofifa.net/teams/10/60.png   

                             club_flag_url  \
0      https://cdn.sofifa.net/flags/fr.png   
1      https://cdn.sofifa.net/flags/de.png   
2  https://cdn.sofifa.net/flags/gb-eng.png   
3      https://cdn.sofifa.net/flags/fr.png   
4  https://cdn.sofifa.net/flags/gb-eng.png   

                            nation_logo_url  \
0  https://cdn.sofifa.net/teams/1369/60.png   
1  https://cdn.sofifa.net/teams/1353/60.png   
2  https://cdn.sofifa.net/teams/1354/60.png   
3                                       NaN   
4  https://cdn.sofifa.net/teams/1325/60.png   

                       nation_flag_url  
0  https://cdn.sofifa.net/flags/ar.png  
1  https://cdn.sofifa.net/flags/pl.png  
2  https://cdn.sofifa.net/flags/pt.png  
3  https://cdn.sofifa.net/flags/br.png  
4  https://cdn.sofifa.net/flags/be.png  

[5 rows x 110 columns]

At first glance, there seems to be a bunch of columns missing (truncated) from the `head()` output. Let's see how many columns we should have by running `df.shape`

In [4]:
df.shape

(19239, 110)

`df.shape` tells us how many rows and how may columns are in our dataframe. We are definitely not seeing all 110 columns in the `head()` output. That's ok, we can set it to show all columns by using `pd.set_option()`.

In [5]:
pd.set_option('display.max_columns', None)

Let's see if it worked...

In [6]:
df.head()

sofifa_id                                         player_url  \
0     158023  https://sofifa.com/player/158023/lionel-messi/...   
1     188545  https://sofifa.com/player/188545/robert-lewand...   
2      20801  https://sofifa.com/player/20801/c-ronaldo-dos-...   
3     190871  https://sofifa.com/player/190871/neymar-da-sil...   
4     192985  https://sofifa.com/player/192985/kevin-de-bruy...   

          short_name                            long_name player_positions  \
0           L. Messi       Lionel Andrés Messi Cuccittini       RW, ST, CF   
1     R. Lewandowski                   Robert Lewandowski               ST   
2  Cristiano Ronaldo  Cristiano Ronaldo dos Santos Aveiro           ST, LW   
3          Neymar Jr        Neymar da Silva Santos Júnior          LW, CAM   
4       K. De Bruyne                      Kevin De Bruyne          CM, CAM   

   overall  potential    value_eur  wage_eur  age         dob  height_cm  \
0       93         93   78000000.0  320000.0   34  1987-06-24        170   
1       92         92  119500000.0  270000.0   32  1988-08-21        185   
2       91         91   45000000.0  270000.0   36  1985-02-05        187   
3       91         91  129000000.0  270000.0   29  1992-02-05        175   
4       91         91  125500000.0  350000.0   30  1991-06-28        181   

   weight_kg  club_team_id            club_name             league_name  \
0         72          73.0  Paris Saint-Germain          French Ligue 1   
1         81          21.0    FC Bayern München    German 1. Bundesliga   
2         83          11.0    Manchester United  English Premier League   
3         68          73.0  Paris Saint-Germain          French Ligue 1   
4         70          10.0      Manchester City  English Premier League   

   league_level club_position  club_jersey_number club_loaned_from  \
0           1.0            RW                30.0              NaN   
1           1.0            ST                 9.0              NaN   
2           1.0            ST                 7.0              NaN   
3           1.0            LW                10.0              NaN   
4           1.0           RCM                17.0              NaN   

  club_joined  club_contract_valid_until  nationality_id nationality_name  \
0  2021-08-10                     2023.0              52        Argentina   
1  2014-07-01                     2023.0              37           Poland   
2  2021-08-27                     2023.0              38         Portugal   
3  2017-08-03                     2025.0              54           Brazil   
4  2015-08-30                     2025.0               7          Belgium   

   nation_team_id nation_position  nation_jersey_number preferred_foot  \
0          1369.0              RW                  10.0           Left   
1          1353.0              RS                   9.0          Right   
2          1354.0              ST                   7.0          Right   
3             NaN             NaN                   NaN          Right   
4          1325.0             RCM                   7.0          Right   

   weak_foot  skill_moves  international_reputation    work_rate body_type  \
0          4            4                         5   Medium/Low    Unique   
1          4            4                         5  High/Medium    Unique   
2          4            5                         5     High/Low    Unique   
3          5            5                         5  High/Medium    Unique   
4          5            4                         4    High/High    Unique   

  real_face  release_clause_eur  \
0       Yes         144300000.0   
1       Yes         197200000.0   
2       Yes          83300000.0   
3       Yes         238700000.0   
4       Yes         232200000.0   

                                         player_tags  \
0  #Dribbler, #Distance Shooter, #FK Specialist, ...   
1  #Aerial Threat, #Distance Shooter, #Clinical F...   
2  #Aerial Threat, #Dribbler, #Distance Shoote

That looks more like 110 columns, however it is hard to read all the column names. Let's use `df.info()`, it will show us all the column names, their index, and also their datatype. This will come in handy later.

In [7]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19239 entries, 0 to 19238
Data columns (total 110 columns):
 #    Column                       Dtype  
---   ------                       -----  
 0    sofifa_id                    int64  
 1    player_url                   object 
 2    short_name                   object 
 3    long_name                    object 
 4    player_positions             object 
 5    overall                      int64  
 6    potential                    int64  
 7    value_eur                    float64
 8    wage_eur                     float64
 9    age                          int64  
 10   dob                          object 
 11   height_cm                    int64  
 12   weight_kg                    int64  
 13   club_team_id                 float64
 14   club_name                    object 
 15   league_name                  object 
 16   league_level                 float64
 17   club_position                object 
 18   club_jersey_number      

## What is a Pandas DataFrame made of?

A DataFrame is made up of both rows and columns. Each column has an index and a name. Looking at the `df.info` output above, we can see the first column has an index of 0 and the name is "sofifa_id".

## Selecting Columns by Name

Using our list of column names above, we can easily return a single column.

In [8]:
name1_df = df.short_name

name1_df

0                  L. Messi
1            R. Lewandowski
2         Cristiano Ronaldo
3                 Neymar Jr
4              K. De Bruyne
                ...        
19234             Song Defu
19235             C. Porter
19236              N. Logue
19237             L. Rudden
19238    E. Lalchhanchhuaha
Name: short_name, Length: 19239, dtype: object

In [9]:
name2_df = df['short_name']

name2_df

0                  L. Messi
1            R. Lewandowski
2         Cristiano Ronaldo
3                 Neymar Jr
4              K. De Bruyne
                ...        
19234             Song Defu
19235             C. Porter
19236              N. Logue
19237             L. Rudden
19238    E. Lalchhanchhuaha
Name: short_name, Length: 19239, dtype: object

These two methods are simple enough, however we were hoping to create new DataFrames. Let's use `type()` to check our new objects type.

In [10]:
type(name1_df)

pandas.core.series.Series

In [11]:
type(name2_df)

pandas.core.series.Series

We can see that both methods have actually created a Pandas Series `pandas.core.series.Series`. In order to create a single column DataFrame, we need to use double brackets.

In [12]:
name3_df = df[['short_name']]

name3_df

short_name
0                L. Messi
1          R. Lewandowski
2       Cristiano Ronaldo
3               Neymar Jr
4            K. De Bruyne
...                   ...
19234           Song Defu
19235           C. Porter
19236            N. Logue
19237           L. Rudden
19238  E. Lalchhanchhuaha

[19239 rows x 1 columns]

Now when we run `type()` we should get a DataFrame.

In [13]:
type(name3_df)

pandas.core.frame.DataFrame

We can use this same syntax to select multiple columns

In [14]:
name_club_age_df = df[['short_name', 'club_name', 'age']]

display(name_club_age_df)

print("Type: ",type(name_club_age_df))

short_name            club_name  age
0                L. Messi  Paris Saint-Germain   34
1          R. Lewandowski    FC Bayern München   32
2       Cristiano Ronaldo    Manchester United   36
3               Neymar Jr  Paris Saint-Germain   29
4            K. De Bruyne      Manchester City   30
...                   ...                  ...  ...
19234           Song Defu             Wuhan FC   22
19235           C. Porter           Derry City   19
19236            N. Logue           Finn Harps   21
19237           L. Rudden           Finn Harps   19
19238  E. Lalchhanchhuaha  NorthEast United FC   19

[19239 rows x 3 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


A benefit of stipulating each column by name is the ability to re-order columns. 

In [15]:
name_age_club_df = df[['short_name', 'age', 'club_name']]

display(name_age_club_df)

print("Type: ",type(name_age_club_df))

short_name  age            club_name
0                L. Messi   34  Paris Saint-Germain
1          R. Lewandowski   32    FC Bayern München
2       Cristiano Ronaldo   36    Manchester United
3               Neymar Jr   29  Paris Saint-Germain
4            K. De Bruyne   30      Manchester City
...                   ...  ...                  ...
19234           Song Defu   22             Wuhan FC
19235           C. Porter   19           Derry City
19236            N. Logue   21           Finn Harps
19237           L. Rudden   19           Finn Harps
19238  E. Lalchhanchhuaha   19  NorthEast United FC

[19239 rows x 3 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


## Selecting Columns by Data Type

You may find yourself in a position where you wish to create a new DataFrame using only columns of a particular data type such as integers or floats. Let's run `df.info` again to see the data types of all our columns.

In [16]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19239 entries, 0 to 19238
Data columns (total 110 columns):
 #    Column                       Dtype  
---   ------                       -----  
 0    sofifa_id                    int64  
 1    player_url                   object 
 2    short_name                   object 
 3    long_name                    object 
 4    player_positions             object 
 5    overall                      int64  
 6    potential                    int64  
 7    value_eur                    float64
 8    wage_eur                     float64
 9    age                          int64  
 10   dob                          object 
 11   height_cm                    int64  
 12   weight_kg                    int64  
 13   club_team_id                 float64
 14   club_name                    object 
 15   league_name                  object 
 16   league_level                 float64
 17   club_position                object 
 18   club_jersey_number      

The output tells us we have 16 columns that are float64, 44 columns that are int64 and 50 columns that are object type. We can use `select_dtypes` to choose columns based on their data types.

In [17]:
float64_df = df.select_dtypes('float64')

display(float64_df)

print("Type: ", type(float64_df))

value_eur  wage_eur  club_team_id  league_level  club_jersey_number  \
0       78000000.0  320000.0          73.0           1.0                30.0   
1      119500000.0  270000.0          21.0           1.0                 9.0   
2       45000000.0  270000.0          11.0           1.0                 7.0   
3      129000000.0  270000.0          73.0           1.0                10.0   
4      125500000.0  350000.0          10.0           1.0                17.0   
...            ...       ...           ...           ...                 ...   
19234      70000.0    1000.0      112541.0           1.0                36.0   
19235     110000.0     500.0         445.0           1.0                27.0   
19236     100000.0     500.0      111131.0           1.0                31.0   
19237     110000.0     500.0      111131.0           1.0                12.0   
19238     110000.0     500.0      113040.0           1.0                13.0   

       club_contract_valid_until  nation_team_id  nation_jersey_number  \
0                         2023.0          1369.0                  10.0   
1                         2023.0          1353.0                   9.0   
2                         2023.0          1354.0                   7.0   
3                         2025.0             NaN                   NaN   
4                         2025.0          1325.0                   7.0   
...                          ...             ...                   ...   
19234                     2021.0             NaN                   NaN   
19235                     2021.0             NaN                   NaN   
19236                     2021.0             NaN                   NaN   
19237                     2021.0             NaN                   NaN   
19238                     2025.0             NaN                   NaN   

       release_clause_eur  pace  shooting  passing  dribbling  defending  \
0             144300000.0  85.0      92.0     91.0       95.0       34.0   
1             197200000.0  78.0      92.0     79.0       86.0       44.0   
2              83300000.0  87.0      94.0     80.0       88.0       34.0   
3             238700000.0  91.0      83.0     86.0       94.0       37.0   
4             232200000.0  76.0      86.0     93.0       88.0       64.0   
...                   ...   ...       ...      ...        ...        ...   
19234            114000.0  58.0      35.0     46.0       48.0       42.0   
19235            193000.0  59.0      39.0     50.0       46.0       41.0   
19236            175000.0  60.0      37.0     45.0       49.0       41.0   
19237            239000.0  68.0      46.0     36.0       48.0       15.0   
19238            217000.0  68.0      38.0     45.0       48.0       36.0   

       physic  goalkeeping_speed  
0        65.0                NaN  
1        82.0                NaN  
2        75.0                NaN  
3        63.0                NaN  
4        78.0                NaN  
...       ...                ...  
19234    49.0                NaN  
19235    51.0                NaN  
19236    52.0                NaN  
19237    42.0                NaN  
19238    48.0                NaN  

[19239 rows x 16 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


In [18]:
object_df = df.select_dtypes('object')

display(object_df)

print("Type: ", type(object_df))

player_url          short_name  \
0      https://sofifa.com/player/158023/lionel-messi/...            L. Messi   
1      https://sofifa.com/player/188545/robert-lewand...      R. Lewandowski   
2      https://sofifa.com/player/20801/c-ronaldo-dos-...   Cristiano Ronaldo   
3      https://sofifa.com/player/190871/neymar-da-sil...           Neymar Jr   
4      https://sofifa.com/player/192985/kevin-de-bruy...        K. De Bruyne   
...                                                  ...                 ...   
19234  https://sofifa.com/player/261962/defu-song/220002           Song Defu   
19235  https://sofifa.com/player/262040/caoimhin-port...           C. Porter   
19236  https://sofifa.com/player/262760/nathan-logue/...            N. Logue   
19237  https://sofifa.com/player/262820/luke-rudden/2...           L. Rudden   
19238  https://sofifa.com/player/264540/emanuel-lalch...  E. Lalchhanchhuaha   

                                 long_name player_positions         dob  \
0           Lionel Andrés Messi Cuccittini       RW, ST, CF  1987-06-24   
1                       Robert Lewandowski               ST  1988-08-21   
2      Cristiano Ronaldo dos Santos Aveiro           ST, LW  1985-02-05   
3            Neymar da Silva Santos Júnior          LW, CAM  1992-02-05   
4                          Kevin De Bruyne          CM, CAM  1991-06-28   
...                                    ...              ...         ...   
19234                                  宋德福              CDM  1999-01-02   
19235                      Caoimhin Porter               CM  2002-03-18   
19236              Nathan Logue-Cunningham               CM  2000-02-07   
19237                          Luke Rudden               ST  2002-04-23   
19238              Emanuel Lalchhanchhuaha              CAM  2002-02-06   

                 club_name                     league_name club_position  \
0      Paris Saint-Germain                  French Ligue 1            RW   
1        FC Bayern München            German 1. Bundesliga            ST   
2        Manchester United          English Premier League            ST   
3      Paris Saint-Germain                  French Ligue 1            LW   
4          Manchester City          English Premier League           RCM   
...                    ...                             ...           ...   
19234             Wuhan FC            Chinese Super League           SUB   
19235           Derry City  Rep. Ireland Airtricity League           RES   
19236           Finn Harps  Rep. Ireland Airtricity League           RES   
19237           Finn Harps  Rep. Ireland Airtricity League           RES   
19238  NorthEast United FC             Indian Super League           SUB   

      club_loaned_from club_joined     nationality_name nation_position  \
0                  NaN  2021-08-10            Argentina              RW   
1                  NaN  2014-07-01               Poland              RS   
2                  NaN  2021-08-27             Portugal              ST   
3                  NaN  2017-08-03               Brazil             NaN   
4                  NaN  2015-08-30              Belgium             RCM   
...                ...         ...                  ...             ...   
19234              NaN  2021-04-01             China PR             NaN   
19235              NaN  2021-02-22  Republic of Ireland             NaN   
19236              NaN  2021-06-18  Republic of Ireland             NaN   
19237              NaN  2021-06-18  Republic of Ireland             NaN   
19238              NaN  2021-08-31                India             NaN   

      preferred_foot      work_rate       body_type real_face  \
0               Left     Medium/Low          Unique       Yes   
1              Right    High/Medium          Unique       Yes   
2              Right       High/Low          Unique       Yes   
3              Right    High/Medium          Unique       Yes   
4              Right      High/High          

Type:  <class 'pandas.core.frame.DataFrame'>


In [19]:
int64_df = df.select_dtypes('int64')

display(int64_df)

print("Type: ", type(int64_df))

sofifa_id  overall  potential  age  height_cm  weight_kg  \
0         158023       93         93   34        170         72   
1         188545       92         92   32        185         81   
2          20801       91         91   36        187         83   
3         190871       91         91   29        175         68   
4         192985       91         91   30        181         70   
...          ...      ...        ...  ...        ...        ...   
19234     261962       47         52   22        180         64   
19235     262040       47         59   19        175         70   
19236     262760       47         55   21        178         72   
19237     262820       47         60   19        173         66   
19238     264540       47         60   19        167         61   

       nationality_id  weak_foot  skill_moves  international_reputation  \
0                  52          4            4                         5   
1                  37          4            4                         5   
2                  38          4            5                         5   
3                  54          5            5                         5   
4                   7          5            4                         4   
...               ...        ...          ...                       ...   
19234             155          3            2                         1   
19235              25          3            2                         1   
19236              25          3            2                         1   
19237              25          3            2                         1   
19238             159          3            2                         1   

       attacking_crossing  attacking_finishing  attacking_heading_accuracy  \
0                      85                   95                          70   
1                      71                   95                          90   
2                      87                   95                          90   
3                      85                   83                          63   
4                      94                   82                          55   
...                   ...                  ...                         ...   
19234                  46                   32                          48   
19235                  54                   33                          46   
19236                  39                   32                          43   
19237                  29                   49                          40   
19238                  39                   32                          46   

       attacking_short_passing  attacking_volleys  skill_dribbling  \
0                           91                 88               96   
1                           85                 89               85   
2                           80                 86               88   
3                           86                 86               95   
4                           94                 82               88   
...                        ...                ...              ...   
19234                       50                 30               45   
19235                       51                 32               41   
19236                       49                 37               47   
19237                       38                 34               42   
19238                       50                 37               48   

       skill_curve  skill_fk_accuracy  skill_long_passing  skill_ball_control  \
0               93                 94                  91                  96   
1               79                 85                  70                  88   
2               81                 84                  77                  88   
3               88                 87                  81                  95   
4               85                 83                  93                  91   
...            ...                ...

Type:  <class 'pandas.core.frame.DataFrame'>


## Selecting Columns Using loc

`loc` is a great way to select row and columns. The syntax can look confusing at first, but let's break it down.


`df.loc[a:b, x:y]` 

The first argument `a:b` refers to the rows you wish to select. As we are focussing on selecting columns in this notebook, we will use `:` which returns all rows.


The second argument `x:y` refers to the columns you want to select. There are three main ways we can use this. We can :


1. Select a single column `df.loc[:,['short_name']]`
2. Select multiple columns `df.loc[:,['short_name', 'overall', 'value_eur']]`
3. Select a slice of columns `df.loc[:,['short_name':'age']`


The slice method will select the two columns and any other that are inbetween. 

In [20]:
name_df = df.loc[:, ['short_name']]

display(name_df)

print("Type: ", type(name_df))

short_name
0                L. Messi
1          R. Lewandowski
2       Cristiano Ronaldo
3               Neymar Jr
4            K. De Bruyne
...                   ...
19234           Song Defu
19235           C. Porter
19236            N. Logue
19237           L. Rudden
19238  E. Lalchhanchhuaha

[19239 rows x 1 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


In [21]:
name_overall_value_df = df.loc[:,['short_name', 'overall', 'value_eur']]

display(name_overall_value_df)

print("Type: ", type(name_overall_value_df))

short_name  overall    value_eur
0                L. Messi       93   78000000.0
1          R. Lewandowski       92  119500000.0
2       Cristiano Ronaldo       91   45000000.0
3               Neymar Jr       91  129000000.0
4            K. De Bruyne       91  125500000.0
...                   ...      ...          ...
19234           Song Defu       47      70000.0
19235           C. Porter       47     110000.0
19236            N. Logue       47     100000.0
19237           L. Rudden       47     110000.0
19238  E. Lalchhanchhuaha       47     110000.0

[19239 rows x 3 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


In [22]:
name_age_slice_df = df.loc[:,'short_name':'age']

display(name_age_slice_df)

print("Type: ", type(name_age_slice_df))

short_name                            long_name  \
0                L. Messi       Lionel Andrés Messi Cuccittini   
1          R. Lewandowski                   Robert Lewandowski   
2       Cristiano Ronaldo  Cristiano Ronaldo dos Santos Aveiro   
3               Neymar Jr        Neymar da Silva Santos Júnior   
4            K. De Bruyne                      Kevin De Bruyne   
...                   ...                                  ...   
19234           Song Defu                                  宋德福   
19235           C. Porter                      Caoimhin Porter   
19236            N. Logue              Nathan Logue-Cunningham   
19237           L. Rudden                          Luke Rudden   
19238  E. Lalchhanchhuaha              Emanuel Lalchhanchhuaha   

      player_positions  overall  potential    value_eur  wage_eur  age  
0           RW, ST, CF       93         93   78000000.0  320000.0   34  
1                   ST       92         92  119500000.0  270000.0   32  
2               ST, LW       91         91   45000000.0  270000.0   36  
3              LW, CAM       91         91  129000000.0  270000.0   29  
4              CM, CAM       91         91  125500000.0  350000.0   30  
...                ...      ...        ...          ...       ...  ...  
19234              CDM       47         52      70000.0    1000.0   22  
19235               CM       47         59     110000.0     500.0   19  
19236               CM       47         55     100000.0     500.0   21  
19237               ST       47         60     110000.0     500.0   19  
19238              CAM       47         60     110000.0     500.0   19  

[19239 rows x 8 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


## Selecting Columns using iloc

`iloc` is very similar to `loc`, however instead of using column names, we can use the columns index. The same three methods can be achieved using `iloc`

In [23]:
iloc_name_df = df.iloc[:,[2]]

display(iloc_name_df)

print("Type: ", type(iloc_name_df))

short_name
0                L. Messi
1          R. Lewandowski
2       Cristiano Ronaldo
3               Neymar Jr
4            K. De Bruyne
...                   ...
19234           Song Defu
19235           C. Porter
19236            N. Logue
19237           L. Rudden
19238  E. Lalchhanchhuaha

[19239 rows x 1 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


In [24]:
iloc_name_overall_value_df = df.iloc[:,[2, 5, 7]]

display(iloc_name_overall_value_df)

print("Type: ", type(iloc_name_overall_value_df))

short_name  overall    value_eur
0                L. Messi       93   78000000.0
1          R. Lewandowski       92  119500000.0
2       Cristiano Ronaldo       91   45000000.0
3               Neymar Jr       91  129000000.0
4            K. De Bruyne       91  125500000.0
...                   ...      ...          ...
19234           Song Defu       47      70000.0
19235           C. Porter       47     110000.0
19236            N. Logue       47     100000.0
19237           L. Rudden       47     110000.0
19238  E. Lalchhanchhuaha       47     110000.0

[19239 rows x 3 columns]

Type:  <class 'pandas.core.frame.DataFrame'>


In [25]:
iloc_name_age_slice = df.iloc[:,2:9]

display(iloc_name_age_slice)

print("Type: ", type(iloc_name_age_slice))

short_name                            long_name  \
0                L. Messi       Lionel Andrés Messi Cuccittini   
1          R. Lewandowski                   Robert Lewandowski   
2       Cristiano Ronaldo  Cristiano Ronaldo dos Santos Aveiro   
3               Neymar Jr        Neymar da Silva Santos Júnior   
4            K. De Bruyne                      Kevin De Bruyne   
...                   ...                                  ...   
19234           Song Defu                                  宋德福   
19235           C. Porter                      Caoimhin Porter   
19236            N. Logue              Nathan Logue-Cunningham   
19237           L. Rudden                          Luke Rudden   
19238  E. Lalchhanchhuaha              Emanuel Lalchhanchhuaha   

      player_positions  overall  potential    value_eur  wage_eur  
0           RW, ST, CF       93         93   78000000.0  320000.0  
1                   ST       92         92  119500000.0  270000.0  
2               ST, LW       91         91   45000000.0  270000.0  
3              LW, CAM       91         91  129000000.0  270000.0  
4              CM, CAM       91         91  125500000.0  350000.0  
...                ...      ...        ...          ...       ...  
19234              CDM       47         52      70000.0    1000.0  
19235               CM       47         59     110000.0     500.0  
19236               CM       47         55     100000.0     500.0  
19237               ST       47         60     110000.0     500.0  
19238              CAM       47         60     110000.0     500.0  

[19239 rows x 7 columns]

Type:  <class 'pandas.core.frame.DataFrame'>
